### PR TITLE
general cleanup of plugins and fixes for TFs

### PR DESCRIPTION
**Pull request by burrimi**
This also closes #247

### CODE REVIEW

**Comment by ffurrer** [comment]
@burrimi can you take a look at this, please?

File: rotors_description/urdf/firefly_generic_odometry_sensor.gazebo [hunk 1]
```diff
@@ -26,15 +26,16 @@
...
-    namespace="${namespace}/odometry_sensor1"
+    namespace="${namespace}"
     odometry_sensor_suffix="1"
     parent_link="${namespace}/base_link"
-    pose_topic="pose"
-    pose_with_covariance_topic="pose_with_covariance"
-    position_topic="position"
-    transform_topic="transform"
-    odometry_topic="odometry"
+    pose_topic="odometry_sensor1/pose"
```
Comment (burrimi) [inline rotors_description/urdf/firefly_generic_odometry_sensor.gazebo]: `pose_topic="odometry_sensor${odometry_sensor_suffix}/pose"` here and below

File: rotors_gazebo/launch/firefly_swarm_hovering_example.launch [hunk 2]
```diff
@@ -100,4 +94,72 @@
     <node name="waypoint_publisher" pkg="rotors_gazebo" type="waypoint_publisher" output="screen" args="-1 0 1 0 10"/>
   </group>
 
+  <group ns="$(arg mav_name)6">
```
Comment (burrimi) [inline rotors_gazebo/launch/firefly_swarm_hovering_example.launch]: do we really need that many mav's?

File: rotors_gazebo/launch/three_multicopters_hovering_example.launch [hunk 3]
```diff
@@ -19,14 +19,16 @@
       <arg name="log_file" value="hummingbird"/>
       <arg name="y" value="0.0"/>
     </include>
-
     <node name="lee_position_controller_node" pkg="rotors_control" type="lee_position_controller_node" output="screen">
       <rosparam command="load" file="$(find rotors_gazebo)/resource/lee_controller_hummingbird.yaml" />
       <rosparam command="load" file="$(find rotors_gazebo)/resource/hummingbird.yaml" />
       <remap from="odometry" to="odometry_sensor1/odometry" />
     </node>
+    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
```
Comment (burrimi) [inline rotors_gazebo/launch/three_multicopters_hovering_example.launch]: add this to the hovering example?

**Comment by burrimi** [comment]
scolgtm.

File: rotors_description/urdf/firefly_generic_odometry_sensor.gazebo (hunk 1 above)
Comment (ffurrer) [inline rotors_description/urdf/firefly_generic_odometry_sensor.gazebo]: Do you want to create a local variable here and then use it in the xacro call (as shown below)?

```XML
<xacro:property name="suffix" value="1" />
<xacro:odometry_plugin_macro
    namespace="${namespace}"
    odometry_sensor_suffix="${suffix}"
    parent_link="${namespace}/base_link"
    pose_topic="odometry_sensor${suffix}/pose"
    ...>
  ...
</xacro:odometry_plugin_macro>
```

File: rotors_gazebo/launch/three_multicopters_hovering_example.launch (hunk 3 above)
Comment (ffurrer) [inline rotors_gazebo/launch/three_multicopters_hovering_example.launch]: done in d0a0d17

**Pull request merged**